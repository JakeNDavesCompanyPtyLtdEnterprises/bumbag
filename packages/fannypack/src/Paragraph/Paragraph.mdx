---
name: Paragraph
route: /typography/paragraph
menu: Typography
---

import { Playground } from 'docz';
import PropsTable from '../_docs/components/PropsTable';
import { Box } from '../primitives/index';
import {Paragraph} from './Paragraph';

# Paragraph

## Import

`import { Paragraph } from 'fannypack'`

## Basic Usage

`<Paragraph>` renders a `<p>` element with reset styles and a bottom margin.

<Playground>
  <Paragraph>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum metus justo, blandit ut leo in, molestie pretium mauris. Pellentesque eros elit, interdum eget mauris et, ultrices auctor nisl. Praesent a dolor vel nulla vehicula lacinia. Proin iaculis ligula nulla.
  </Paragraph>
  <Paragraph>
    Morbi tincidunt cursus leo, sit amet porttitor mi. In a diam volutpat, blandit felis eu, viverra nunc. Nullam molestie iaculis elit, non placerat massa. In et consectetur lorem. Vivamus feugiat iaculis aliquet. Nunc laoreet augue in leo venenatis, id condimentum nisl eleifend.
  </Paragraph>
</Playground>

> Remember, you can also specify different colors with the `color` prop.

<Playground>
  <Paragraph color="primary">
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum metus justo, blandit ut leo in, molestie pretium mauris. Pellentesque eros elit, interdum eget mauris et, ultrices auctor nisl. Praesent a dolor vel nulla vehicula lacinia. Proin iaculis ligula nulla.
  </Paragraph>
</Playground>

## Props

<PropsTable component={Paragraph} />

## Theming

### Schema

```jsx
{
  Paragraph: {
    base: css | Object
  }
}
```

[Learn more about theming](/styling#theming)
