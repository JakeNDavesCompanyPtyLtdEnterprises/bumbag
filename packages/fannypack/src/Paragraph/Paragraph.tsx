import * as React from 'react';
import * as PropTypes from 'prop-types';
import { ParagraphProps as ReakitParagraphProps } from 'reakit/ts';

import _Paragraph from './styled';

export type LocalParagraphProps = {
  children: React.ReactNode;
  className?: string;
};
export type ParagraphProps = ReakitParagraphProps & LocalParagraphProps;

export const Paragraph: React.FunctionComponent<LocalParagraphProps> = ({ children, className, ...props }) => (
  <_Paragraph className={className} {...props}>
    {children}
  </_Paragraph>
);

Paragraph.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};
Paragraph.defaultProps = {
  className: undefined
};

const C: React.FunctionComponent<ParagraphProps> = Paragraph;
export default C;
