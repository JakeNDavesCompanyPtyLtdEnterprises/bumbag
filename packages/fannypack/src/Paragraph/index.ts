export { default as Paragraph } from './Paragraph';
export { default } from './Paragraph';
