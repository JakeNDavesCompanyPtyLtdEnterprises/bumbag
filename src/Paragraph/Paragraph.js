// @flow
import React, { type Node } from 'react';

import type { Palette } from '../types';
import _Paragraph from './styled';

type Props = {
  children: Node,
  className?: string,
  color?: Palette
};

const Paragraph = ({ children, className, ...props }: Props) => (
  <_Paragraph className={className} {...props}>
    {children}
  </_Paragraph>
);

Paragraph.defaultProps = {
  className: undefined,
  color: undefined
};

export default Paragraph;
