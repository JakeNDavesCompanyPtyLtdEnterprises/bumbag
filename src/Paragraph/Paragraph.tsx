import * as React from 'react';
// @ts-ignore
import PropTypes from 'prop-types';
import { ParagraphProps as ReakitParagraphProps } from 'reakit/ts';

import _Paragraph from './styled';

export interface LocalParagraphProps {
  children: React.ReactNode;
  className?: string;
}
export type ParagraphProps = ReakitParagraphProps & LocalParagraphProps;

export const Paragraph: React.FunctionComponent<LocalParagraphProps> = ({ children, className, ...props }) => (
  <_Paragraph className={className} {...props}>
    {children}
  </_Paragraph>
);

Paragraph.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};
Paragraph.defaultProps = {
  className: undefined
};

const C: React.FunctionComponent<ParagraphProps> = Paragraph;
export default Paragraph;
