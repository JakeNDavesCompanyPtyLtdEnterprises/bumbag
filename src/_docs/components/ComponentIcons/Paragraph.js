import React from 'react';

const SvgComponent = props => (
  <svg width={76} height={48} fill="none" {...props}>
    <path
      d="M75.564 8H0v3.649h75.564V8zM75.564 0H0v3.649h75.564V0zM76 16H0v3h76v-3zM75.564 36H0v3.649h75.564V36zM75.564 28H0v3.649h75.564V28zM37.782 43.663H0v3.649h37.782v-3.65z"
      fill="#3926A5"
    />
  </svg>
);

export default SvgComponent;
