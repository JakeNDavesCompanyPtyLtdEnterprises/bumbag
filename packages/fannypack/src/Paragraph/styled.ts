import Paragraph from 'reakit/Paragraph';

import styled, { space, theme, selector } from '../styled';
import Icon from '../Icon/styled';
import { ParagraphProps } from './Paragraph';

export default styled(Paragraph)<ParagraphProps>`
  &:not(:last-child) {
    margin-bottom: ${space(4)}rem;
  }

  & ${selector(Icon)} {
    top: 0.15em;
  }

  & {
    ${theme('fannypack.Paragraph.base')};
  }
`;
