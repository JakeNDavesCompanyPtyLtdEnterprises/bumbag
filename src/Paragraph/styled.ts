// @ts-ignore
import Paragraph from 'reakit/Paragraph';
import { theme } from 'styled-tools';

import styled from '../styled';
import Icon from '../Icon/styled';
import { ParagraphProps } from './Paragraph';

export default styled(Paragraph)<ParagraphProps>`
  &:not(:last-child) {
    margin: 0 0 1rem;
  }

  & ${Icon} {
    top: 0.15em;
  }

  & {
    ${theme('fannypack.Paragraph.base')};
  }
`;
