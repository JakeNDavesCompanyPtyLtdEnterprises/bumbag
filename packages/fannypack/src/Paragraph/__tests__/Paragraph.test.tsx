import * as React from 'react';
import render from '../../_utils/tests/render';
import Paragraph from '../Paragraph';

it('renders correctly for a basic paragraph', () => {
  const { container } = render(<Paragraph>test</Paragraph>);
  expect(container.firstChild).toMatchSnapshot();
});
