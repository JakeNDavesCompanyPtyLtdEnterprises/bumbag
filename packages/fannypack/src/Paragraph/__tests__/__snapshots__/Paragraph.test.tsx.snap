// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for a basic paragraph 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c1:focus:not(:focus-visible) {
  outline: none;
}

.c0:not(:last-child) {
  margin-bottom: 1rem;
}

.c0 .c2 {
  top: 0.15em;
}

<p
  class="c0 c1"
>
  test
</p>
`;
